Assets/_Scripts/Actions/ActionBase.cs

```diff
@@ -193,5 +222,47 @@ public abstract class ActionBase : ScriptableObject {
         return target.targetId;
     }
 
-    public abstract FightResult ApplyEffect(FightingEntity user, List<FightingEntity> targets);
+    protected void InflictStatus(FightingEntity target, AilmentInfliction infliction) {
```

Comment:
Doesn't it make a bit more sense to put this somewhere in AilmentController? (And deprecate AddStatusAIlment)

Reply:
I can see the value in that. ActionBase still needs to know what status ailments were successfully inflicted though. Perhaps more of the infliction responsibility (i.e. rolling the randomness) can be moved to AilmentInfliction or AilmentController?

Reply:
I'm not quite sure what you mean. I'm fine with the chance being done in the action side, but applying the ailment should be done in the AilmentController side, so we can use it for non-action status ailments (i.e. Rewards/passives)

Also, if you want to know if an ailment was inflicted, you can always do ailmentController.GetAilment(name).

Reply:
I've moved InflictStatus code over to AilmentController, but I also left AddStatusAilment alone for instances where we want to apply an ailment directly instead of using an AilmentInfliction. Thoughts?

Reply:
Should be fine for now